codec-http/src/main/java/io/netty/handler/codec/http/multipart/AbstractDiskHttpData.java

```diff
@@ -300,14 +300,20 @@ public ByteBuf getChunk(int length) throws IOException {
         }
         int read = 0;
         ByteBuffer byteBuffer = ByteBuffer.allocate(length);
-        while (read < length) {
-            int readnow = fileChannel.read(byteBuffer);
-            if (readnow == -1) {
+        try {
+            while (read < length) {
+                int readnow = fileChannel.read(byteBuffer);
+                if (readnow == -1) {
+                    fileChannel.close();
+                    fileChannel = null;
+                    break;
+                } else {
+                    read += readnow;
+                }
+            }
+        } finally {
+            if (fileChannel != null) {
```

Comment:
What about this version? Interest: no close within the loop, setting to null in finally (no check needed)

        try {
            while (read < length) {
                int readnow = fileChannel.read(byteBuffer);
                if (readnow == -1) {
                    break;
                }
                read += readnow;
            }
        } finally {
            fileChannel.close();
            fileChannel = null;
        }

Reply:
Thanks